class.jetpack.php

```diff
@@ -7364,6 +7364,12 @@ public static function load_jetpack_gutenberg() {
 			$version
 		);
 
+		wp_localize_script(
+			'jetpack-blocks-editor',
+			'JETPACK_BLOCK_ASSETS_BASE_URL',
+			plugins_url( '_inc/blocks/', JETPACK__PLUGIN_FILE )
```

Comment:
Would it be possible to store this in an array? `wp_localize_script` expects an array here, not a string:
https://developer.wordpress.org/reference/functions/wp_localize_script/

Reply:
I'm torn here. This works fine. What we want to define is a global string, not an array or object.

It's unclear to me whether `wp_localize_script` works with simple strings by accident or design.

Gutenberg uses this pattern to expose strings: https://github.com/WordPress/gutenberg/blob/3d3c2a4d52903131f68d2694bd551408561fa778/lib/client-assets.php#L1393

Initially, I'm inclined to leave this as-is rather than wrap this in an array or object.

I would be surprised if this core functionality had breaking changes. That said, I looked for core tests around this and didn't find any. I'm open to change if folks have strong opinions.

Reply:
> It's unclear to me whether wp_localize_script works with simple strings by accident or design. 

Looks to me like it expects an array, but will force an array if needed.

> I'm open to change if folks have strong opinions.

No strong opinions here. I'll let someone else weigh in on this.

Reply:
PHP will convert any scalar values to array with a single scalar value element (`wp_localize_script` uses a specific `(array)$var` cast). So if we summarize, having a non-array value there:

* + Works well
* + Is simple and straightforward
* - Relies on PHP casting scalars to arrays in a specific way, and WP using that in a specific way
* - Contradicts the `wp_localize_script` function documentation

Expecting an array there makes sense because of the initial purpose of that function - passing localized strings to a script. It's fair to say we're over-engineering the usage this function, but it's also fair to say that it can be used for more than just localizing translatable strings.

That being said, I've used `wp_localize_script()` to expose scalars, and it has worked well. Considering that Gutenberg also does that makes me think it's a common pattern, albeit undocumented. It would probably be a good idea for contributing an enhancement for the core method to allow scalars without casting to array, or to at least add some tests for this case, but that's out of scope. 

It's not a precedent, so I'm good to keep as-is 👍